charts/liquibase-migration/values.yaml

```diff
@@ -1,53 +1,106 @@
-#   DEMONSTRATION: This file is NOT used because this is a library chart.             #
-#   The source of truth is in the templates themselves, this file is documentation.   #
-#                                                                                     #
-#   All unset values are required.                                                    #
-#                                                                                     #
-# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# -- Specific migrations to run; each config merged over the defaults to form the full configuration
+# @default -- None by default, one entry here required per migration
+enumerated:
+  - # -- (string) Required name of this specific migration
+    name:
+    # Keys from the defaults may be given here to override them for this specific migration
 
-vault:
-  # Migration credentials only referenced if migration.enabled == true
-  migration:
-    # vault.migration.path -- (string) Required Vault path to secret containing DB credentials
-    path:
-    # vault.migration.dbUsernameKey -- (string) Required key in Vault secret to DB username
-    dbUsernameKey:
-    # vault.migration.DbPasswordKey -- (string) Required key in Vault secret to DB password
-    dbPasswordKey:
+# -- Baseline config merged under each enumerated one to form the full configuration for that particular migration
+# @default -- See sub-keys
+defaults:
 
-migration:
-  # migration.enabled -- (bool) Whether to run a Liquibase migration during sync
+  # -- Whether to enable migrations by default
   enabled: false
-  # migration.imageName -- (string) Required full app image name, without trailing tag
-  # Like "gcr.io/broad-dsp-gcr-public/rawls"
-  imageName:
-  # migration.jarLocation -- (string) Required jar location in app image, expanded by migration.appContainerShell
-  # Like "$(find /rawls -name 'rawls*.jar')"
-  jarLocation:
-  # migration.secretPrefix -- (string) Required prefix of -app-ctmpls, -sqlproxy-ctmpls, -sqlproxy-env secrets
-  # Like "rawls-backend"
-  secretPrefix:
-  # migration.appName -- (string) Name of the application being deployed, for deriving other values
+  
+  # -- (string) Name of the application being deployed
   # @default -- Set to .Release.Name
-  appName: null
-  # migration.labelRef -- (string) Template reference to `include` in resource labels
-  # @default -- Set to "${appName}.labels"
-  labelRef: null
-  # migration.syncWave -- (string) ArgoCD wave to apply these resources in
-  syncWave: "-1"
-  # migration.imageTag -- (string) Tag of the migration.imageName to use
-  # WARNING: App may behave unexpectedly if its database has been migrated on a different version than the app itself
+  k8sAppName: null
+  # -- (string) Template reference to use to obtain resource labels, set to empty to disable
+  # @default -- Set to `${k8sAppName}.labels`
```

Comment:
Nice, love the use of the downstream chart template here!

Reply:
it actually breaks in annoying ways, which is unfortunate. The root chart's template is re-rendered, which means any .Values reference in the template becomes that of liquibase-migrations. This means you have to do really unfortunate things like [this](https://github.com/broadinstitute/terra-helm/pull/504/files#diff-f0039169139fa8741996dc6770e9d2579a1574c3c0b6c2fc2a36db470af63af0R212) if the root chart uses .Values.name. A fix would be to stop using .Values.name and instead move things to .Values.global.name :/

Reply:
Ah, I guess that's the downside to the application chart approach. Can't pass in specific context like with a library chart.

I guess the library chart does still have some pros:
* can pass in templates by name, with any context at all
* can use merging to override rendered resources, as Eno demoed with the ingress lib chart

The goal with the application chart approach was to have a real `values.yaml` file with defaults right?
I'm wondering if we could simulate that in a library chart with some hacky trick like `$.Files.Get values.yaml | fromYaml`

(Just some rambling thoughts, not convinced either option is better)